src/nni_manager/training_service/local/localTrainingService.ts

```diff
@@ -345,6 +345,17 @@ class LocalTrainingService implements TrainingService {
             const oldStatus: TrialJobStatus = trialJob.status;
             trialJob.status = newStatus;
             this.onTrialJobStatusChanged(trialJob, oldStatus);
+            //if job is not running, destory job stream
+            if(['SUCCEEDED', 'FAILED', 'USER_CANCELED', 'SYS_CANCELED', 'EARLY_STOPPED'].includes(newStatus)) {
+                if(this.jobStreamMap.has(trialJob.id)) {
+                    const stream = this.jobStreamMap.get(trialJob.id);
+                    if(!stream) {
+                        throw new Error(`Could not find stream in trial ${trialJob.id}`);
+                    }
+                    stream.destroy();
+                    this.jobStreamMap.delete(trialJob.id);
```

Comment:
This is not a good place to destroy job stream ,  it is better to put it into onTrialJobStatusChanged.

Reply:
moved to onTrialJobStatusChanged.